zerver/migrations/0513_copy_groups_data_to_named_user_group.py

```diff
@@ -0,0 +1,55 @@
+# Generated by Django 4.2.11 on 2024-04-03 13:36
+
+from django.db import connection, migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from psycopg2.sql import SQL
+
+
+def create_named_user_group_objects_for_groups(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    with connection.cursor() as cursor:
+        cursor.execute(SQL("SELECT MAX(id) FROM zerver_usergroup"))
+        (max_id,) = cursor.fetchone()
+
+    if max_id is None:
+        return
+
+    BATCH_SIZE = 1000
+    lower_id_bound = 0
+    while lower_id_bound < max_id:
+        upper_id_bound = min(lower_id_bound + BATCH_SIZE, max_id)
+        with connection.cursor() as cursor:
+            query = SQL("""
+                INSERT INTO zerver_namedusergroup (usergroup_ptr_id, realm_id, name, description, is_system_group, can_mention_group_id)
+                SELECT id, realm_id, name, description, is_system_group, can_mention_group_id
+                FROM zerver_usergroup
+                WHERE id > %(lower_id_bound)s AND id <= %(upper_id_bound)s;
+                """)
+            cursor.execute(
+                query,
+                {
+                    "lower_id_bound": lower_id_bound,
+                    "upper_id_bound": upper_id_bound,
+                },
+            )
+
+        print(f"Processed {upper_id_bound} / {max_id}")
+        lower_id_bound = lower_id_bound + BATCH_SIZE
```

Comment:
@alexmv can you help review this? This migration looks good to me, though I think we will want to be careful about how we order the deployment; might make sense to just do a temp commit to temporarily freeze the UserGroup API and realm creation APIs (i.e. the things that can create or modify a group), then do this migration in the background, then deploy all the code changes.